### ISSUE TITLE
Improve validation of Singpass/Corppass relay state

### ISSUE THREAD

**Issue by mantariksh** [opengovsg/FormSG#2539]
1. At the point where the relay state is handed back to the /singpass/login and /corppass/login endpoints, the `formId` should be validated to be a valid MongoDB ObjectID the moment it is parsed. Currently the validation is very loose, and it only happens to be validated when we attempt to retrieve the form by its ID.
2. The `DESTINATION_REGEX` should be used with `.match` instead of `.exec`, since `.exec` mutates the state of the regex and the same regex is shared across multiple functions.

**Comment by timotheeg** [comment]
Just to add a small clarification on `exec()` vs `match()` . The mutation is only an issue when regexes has a `g` flag, which is not the case here, but still: because such cases are really not obvious at all, I recommend using `.match()` all the time where the intent is to do a single extraction (or just `.test()`),  and only use `exec()` when you need to traverse a source string with multiple matches.

Sample script to illustrate the issue:
```
s1 = "foobar";
s2 = "foobarbaz";

re = /foo/g;

console.log( !! re.exec(s1) ); // true
console.log( !! re.exec(s2) ); // false
```

**Comment by frankchn** [comment]
Fixed in #2869
